File: cs365/hw5/Stack/StackDriver.java
import Stack.*;
import java.util.Scanner;

class StackDriver {
    public static void main(String args[]) {
	Stack<String> myStack = new Stack<String>(5);
	Stack<String> defaultStack = new Stack<String>();

	Scanner console = new Scanner(System.in);
	Scanner lineTokenizer;
	
	while (console.hasNextLine()) {
	    lineTokenizer = new Scanner(console.nextLine());
	    String cmd = lineTokenizer.next();
	    if (cmd.equals("push")) {
		String value = lineTokenizer.next();
		if (!myStack.push(value)) {
		    System.out.printf("*** Error: Stack full. Cannot push %s onto stack\n", value);
	  	};
	    }
	    else if (cmd.equals("pop")) {
		String value = myStack.pop();
		defaultStack.push(value);
		System.out.println("popped value = " + value);
	    }
	    else 
		System.out.println("bad command: " + cmd);
	    lineTokenizer.close();
	}
	 
	// print the contents of the two stacks
	System.out.println("*** contents of myStack ***");
	while (!myStack.isEmpty()) 
	    System.out.println(myStack.pop());
	System.out.println("*** contents of defaultStack ***");
	while (!defaultStack.isEmpty()) 
	    System.out.println(defaultStack.pop());
    }
}

File: cs365/hw5/Stack/StackDriver.cpp
#include <iostream>
#include "Stack.h"

using namespace std;

int main() {
  Stack<string,5> myStack;
  Stack<string> defaultStack;
  string cmd, value;

  while (cin >> cmd) {
    if (cmd == "push") {
      cin >> value;
      if (!myStack.push(value)) {
	cout << "*** Error: Stack full. Cannot push " << value << " onto stack\n";
      }
    }
    else if (cmd == "pop") {
      value = myStack.pop();
      defaultStack.push(value);
      cout << "popped value = " << value << endl;
    }
    else 
      cout <<"bad command: " << cmd << endl;
  }
	 
  // print the contents of the two stacks
  cout << "*** contents of myStack ***" << endl;
  while (!myStack.isEmpty())
    cout << myStack.pop() << endl;

  cout << "*** contents of defaultStack ***" << endl;
  while (!defaultStack.isEmpty()) 
    cout << defaultStack.pop() << endl;
}
